### PR TITLE
feat: ability to save and retrieve the explorer queries

### LINKED ISSUE #2242: (explorer-queries): ability to save and fetch the explorer query information
By moving this part to the backend, we reduce the effort on the frontend to maintain the global state, event emitter, parser etc... Giving the client ability to save the query from the explorer and use the reference ID to fetch data back in the alerts page. This is going to be same in all the pages including the dashboards.

cc // @palashgdev

### PR DESCRIPTION
**Pull request by srikanthccv**
Fixes #2242

### CODE REVIEW

**Review by palashgdev — APPROVED** [review]

File: pkg/query-service/app/http_handler.go [hunk 1]
```diff
@@ -2245,3 +2253,70 @@ func (aH *APIHandler) logAggregate(w http.ResponseWriter, r *http.Request) {
...
+		RespondError(w, &model.ApiError{Typ: model.ErrorInternal, Err: err}, nil)
+		return
+	}
+	aH.Respond(w, queries)
+}
+
+func (aH *APIHandler) createExplorerQueries(w http.ResponseWriter, r *http.Request) {
+	var query v3.ExplorerQuery
+	err := json.NewDecoder(r.Body).Decode(&query)
+	if err != nil {
```
Comment (nityanandagohain) [inline pkg/query-service/app/http_handler.go:2269]: lets add some validation for the request payload.

File: pkg/query-service/app/http_handler.go [hunk 2]
```diff
@@ -2245,3 +2253,70 @@ func (aH *APIHandler) logAggregate(w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+
+	aH.Respond(w, query)
+}
+
+func (aH *APIHandler) updateExplorerQuery(w http.ResponseWriter, r *http.Request) {
+	queryID := mux.Vars(r)["queryId"]
+	var query v3.ExplorerQuery
+	err := json.NewDecoder(r.Body).Decode(&query)
```
Comment (nityanandagohain) [inline pkg/query-service/app/http_handler.go:2297]: same as above, some validation will be good

**Comment by srikanthccv** [comment]
I will address the comments once the https://github.com/SigNoz/signoz/pull/2263 get in since it has some relevant bits.

File: pkg/query-service/model/v3/v3.go [hunk 3]
```diff
@@ -238,3 +238,17 @@ type Point struct {
...
+// The source page name is used to identify the page that initiated the query
+// The source page could be "traces", "logs", "metrics" or "dashboards", "alerts" etc.
+type ExplorerQuery struct {
+	UUID           string          `json:"uuid,omitempty"`
+	SourcePage     string          `json:"sourcePage"`
+	CompositeQuery *CompositeQuery `json:"compositeQuery"`
+	// ExtraData is JSON encoded data used by frontend to store additional data
+	ExtraData string `json:"extraData"`
+	// 0 - false, 1 - true
+	IsView int8 `json:"isView"`
```
Comment (makeavish) [inline pkg/query-service/model/v3/v3.go:253]: What is the use-case of `IsView`?

File: pkg/query-service/app/explorer/db.go [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+func InitWithDSN(dataSourceName string) (*sqlx.DB, error) {
+	var err error
+
+	db, err = sqlx.Open("sqlite3", dataSourceName)
+	if err != nil {
+		return nil, err
+	}
+
+	tableSchema := `CREATE TABLE IF NOT EXISTS explorer_queries (
+		id INTEGER PRIMARY KEY AUTOINCREMENT,
```
Comment (makeavish) [inline pkg/query-service/app/explorer/db.go:37]: Why not use `uuid` as primary key?

File: pkg/query-service/model/v3/v3.go (hunk 3 above)
Comment (srikanthccv) [reply pkg/query-service/model/v3/v3.go:253]: >Possibility to save/load filters in Traces and Logs . This view has a lot of features. Just i would like to have possibility to save it . And reuse again later with actual timeframe

https://github.com/SigNoz/signoz/issues/1958
https://github.com/SigNoz/signoz/issues/2172

File: pkg/query-service/app/explorer/db.go (hunk 4 above)
Comment (srikanthccv) [reply pkg/query-service/app/explorer/db.go:37]: We could do that as well. I don't remember why I did this. Will update.

File: pkg/query-service/app/http_handler.go (hunk 2 above)
Comment (srikanthccv) [reply pkg/query-service/app/http_handler.go:2307]: Updated

File: pkg/query-service/app/explorer/db.go (hunk 4 above)
Comment (srikanthccv) [reply pkg/query-service/app/explorer/db.go]: uuid is primary key now

**Review by nityanandagohain — APPROVED** [review]
LGTM

**Pull request merged**